### PR TITLE
[14.0][FIX] upgrade_analysis: fix methods related to noupdate data

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Last time I did the analysis I detected that the noupdate data code was broken and thus, not updated. The break of that is silenced, so I didn't see it. But with more time, I have been fixing that to make it works. It will also be needed https://github.com/OCA/OpenUpgrade/pull/2632 compatibility to make it work. The result is https://github.com/OCA/OpenUpgrade/pull/2631.

Errors fixed:
- ~~Method `list_modules` was returning a list of lists, which was wrong.~~
- Method `get_xml_records` was reading the files to get parsed later with `_parse_files`. That breaks when the xml file uses a different enconding than `utf-8`, like `data/account_chart_template_data.xml` in `l10n_ma`. In fact, the reading step proves to be unnecessary, as it can be parsed directly, that's what it does now (and what it was already doing in the script of v13).
- Also merged_modules of apriori was not being considered.

And then, I update my company name :P

~~Note: I also added the analysis.txt to justify the migration scripts haha~~

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

File: upgrade_analysis/models/upgrade_record.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def list_modules(self):
             """SELECT DISTINCT(module) FROM upgrade_record
             ORDER BY module"""
         )
-        return [module for module, in self.env.cr.fetchall()]
+        return [module[0] for module in self.env.cr.fetchall()]
```
Comment (StefanRijnhart) [inline upgrade_analysis/models/upgrade_record.py]: Why did you change this? I think the original is a very elegant way to parse tuples of size one.

File: upgrade_analysis/migrations/14.0.1.0.0/upgrade_analysis.txt [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---Models in module 'upgrade_analysis'---
```
Comment (StefanRijnhart) [inline upgrade_analysis/migrations/14.0.1.0.0/upgrade_analysis.txt]: Do we want to include the analysis of non-core modules? I would prefer not to.

File: upgrade_analysis/models/upgrade_record.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply upgrade_analysis/models/upgrade_record.py]: No, without that `[0]` then you return something like [("module1",), ("module2",), ...] which later breaks. But if you return instead ["module1", "module2", ...] then later it doesn't break. My changes have been tested.

File: upgrade_analysis/migrations/14.0.1.0.0/upgrade_analysis.txt (hunk 2 above)
Comment (MiquelRForgeFlow) [reply upgrade_analysis/migrations/14.0.1.0.0/upgrade_analysis.txt]: Ok, then I will remove then.

File: upgrade_analysis/models/upgrade_record.py (hunk 1 above)
Comment (StefanRijnhart) [reply upgrade_analysis/models/upgrade_record.py]: The comma after `module` unpacks the tuple fine.
```
>>> [module for module, in  [("module1",), ("module2",),]]
['module1', 'module2']
```

Comment (MiquelRForgeFlow) [reply upgrade_analysis/models/upgrade_record.py]: Oh, shit, then it was my bad. Yeah, it's elegant, but where does it fall the readability? hehe

**Review by StefanRijnhart — APPROVED** [review]
